src/corredor/helpers/messaging.ts

```diff
@@ -1,15 +1,32 @@
 /* eslint-disable */
 
-import { Channel } from '../../messaging/types/channel'
-import { User } from '../../system/types/user'
-import { extractID, genericPermissionUpdater, isFresh } from './shared'
+import { extractID, genericPermissionUpdater, isFresh, Rule } from './shared'
+import { Messaging as MessagingAPI } from '../../api-clients'
+import { Channel, Message } from '../../messaging/'
+import { User } from '../../system'
+
+type StrIndex<TValue> = {
```

Comment:
Why the `Str` prefix? And this is more a map then an index :)
Plus, try to follow standard naming conventions for generic types (TValue => T)

But do read the rest of the comments -- I hope you'll find a way to get rid of this at the end